libs/SalesforceSDKCore/SalesforceSDKCore/Classes/RestAPI/SFRestAPI.m

```diff
@@ -371,24 +479,35 @@ - (SFRestRequest *) compositeRequest:(NSArray<SFRestRequest*>*) requests refIds:
         NSMutableDictionary<NSString *, id> *requestJson = [NSMutableDictionary new];
         requestJson[@"referenceId"] = refId;
         requestJson[@"method"] = [SFRestRequest httpMethodFromSFRestMethod:request.method];
+
         // queryParams belong in url
         if (request.method == SFRestMethodGET || request.method == SFRestMethodDELETE) {
             requestJson[@"url"] = [NSString stringWithFormat:@"%@%@%@", request.endpoint, request.path, [self toQueryString:request.queryParams]];
         }
+
         // queryParams belongs in body
         else {
             requestJson[@"url"] = [NSString stringWithFormat:@"%@%@", request.endpoint, request.path];
-            requestJson[@"body"] = request.queryParams;
+            requestJson[@"body"] = request.requestBodyAsDictionary;
         }
         [requestsArrayJson addObject:requestJson];
     }
     NSMutableDictionary<NSString *, id> *compositeRequestJson = [NSMutableDictionary new];
     compositeRequestJson[@"compositeRequest"] = requestsArrayJson;
     compositeRequestJson[@"allOrNone"] = [NSNumber numberWithBool:allOrNone];
-
     NSString *path = [NSString stringWithFormat:@"/%@/composite", self.apiVersion];
-
-    return [SFRestRequest requestWithMethod:SFRestMethodPOST path:path queryParams:compositeRequestJson];
+    SFRestRequest *request = [SFRestRequest requestWithMethod:SFRestMethodPOST path:path queryParams:nil];
+    NSError *error = nil;
+    if (compositeRequestJson) {
+        request.requestBodyAsDictionary = compositeRequestJson;
```

Comment:
Shouldn't SFRestRequest takes care of creating its custom request body data when the requestBodyAsDictionary is provided?
Maybe we could have a new constructor for SFRestRequest that accepts the body as dictionary and takes care of building the request body (like we do on android: https://github.com/forcedotcom/SalesforceMobileSDK-Android/blob/dev/libs/SalesforceSDK/src/com/salesforce/androidsdk/rest/RestRequest.java#L239)